coalib/misc/Caching.py

```diff
@@ -0,0 +1,81 @@
+import calendar
+import time
+import hashlib
+
+from coalib.output.Tagging import get_tags_dir as get_pickle_dir
+from coalib.misc.CachingUtilities import (
+    pickle_load, pickle_dump, time_consistent, update_time_db)
+
+
+class Cache:
```

Comment:
not sure about moving out but keeping it generic is good because it already is, functionally at least